### ISSUE TITLE
Add projection for the hyperbolic space

### ISSUE THREAD

**Issue by mateuszbaran** [JuliaNLSolvers/Manifolds.jl#79]
As discussed here: https://github.com/JuliaNLSolvers/Manifolds.jl/pull/71#issuecomment-569412815

**Comment by kellertuer** [comment]
I would still like to see a reference /literature for such a projection or a verification that your proposed procedure _is_ the closest point on the manifold to said point from the embedding.

**Comment by mateuszbaran** [comment]
Right, I'd like to verify it too. I tried to derive it, it looks deceptively simple since there are very nice symmetries but the Minkowski distance is so unintuitive 🙁 .

**Comment by kellertuer** [comment]
...and intuitively, I even think your formula is right ;)

**Comment by kellertuer** [comment]
I checked the reference again (see https://math.stackexchange.com/questions/3175081/projection-of-point-onto-closest-point-on-geodesic-in-hyperbolic-geometry-hyper/3176477#3176477) and it refers to projection a point on the hyperboloid model onto a geodesic on the hyperboloid model, so its not the `project(M,x)`for a point from the embedding we are looking for. Actually there is a thread ( https://math.stackexchange.com/questions/3218487/orthogonal-projection-of-point-in-ambient-space-onto-hyperboloid ), so I am not sure whether there exists a formula this issue is looking for?
